src/collectors/hadoop/hadoop.py

```diff
@@ -44,7 +44,12 @@ def get_default_config(self):
         return config
 
     def collect(self):
-        for pattern in self.config['metrics']:
+        if isinstance(self.config['metrics'], list):
```

Comment:
Yes, you are right -- this is the better way to do it. I've made the changes and tested them and it works just fine as far as I can tell.